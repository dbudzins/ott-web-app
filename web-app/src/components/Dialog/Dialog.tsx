import React from 'react';

import Modal from '../Modal/Modal';
import Slide from '../Animation/Slide/Slide';
import ModalCloseButton from '../ModalCloseButton/ModalCloseButton';

import styles from './Dialog.module.scss';

type Props = {
  open: boolean;
  onClose: () => void;
  children: React.ReactNode;
};

const Dialog: React.FC<Props> = ({ open, onClose, children }: Props) => {
  return (
    <Modal open={open} onClose={onClose} AnimationComponent={Slide}>
      <div className={styles.dialog}>
        <ModalCloseButton onClick={onClose} />
        {children}
      </div>
    </Modal>
  );
};

export default Dialog;
