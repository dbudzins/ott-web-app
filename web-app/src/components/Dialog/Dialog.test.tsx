import React from 'react';
import { render } from '@testing-library/react';

import Dialog from './Dialog';

vi.mock('../Modal/Modal', () => ({
  default: 'div',
}));

describe('<Dialog>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(
      <Dialog onClose={vi.fn()} open>
        Dialog contents
      </Dialog>,
    );

    expect(container).toMatchSnapshot();
  });
});
