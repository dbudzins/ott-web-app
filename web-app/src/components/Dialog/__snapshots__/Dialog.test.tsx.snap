// Vitest Snapshot v1

exports[`<Dialog> > renders and matches snapshot 1`] = `
<div>
  <div
    open=""
  >
    <div
      class="dialog"
    >
      <div
        aria-label="close_modal"
        class="iconButton modalCloseButton"
        role="button"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="icon"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </div>
      Dialog contents
    </div>
  </div>
</div>
`;
